src/services/todo_service.py

```diff
@@ -0,0 +1,151 @@
+
+
+def get_single_todo(task_id):
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+    statement = "SELECT * FROM TASKS WHERE task_id = ?"
+    cursor.execute(statement, (task_id,))
+    task_matching = cursor.fetchall()
+    if len(task_matching) == 0:
+        return make_response(jsonify({"description": f"task '{task_id}' not found"}), 404)
+    else:
+        return make_response(
+            jsonify(tasks_response_builder(task_matching[0]), 200))
+
+
+def delete_single_todo(task_id):
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+    # check if task exist
+    statement = "SELECT * FROM TASKS WHERE task_id = ?"
+    cursor.execute(statement, (task_id,))
+    deleted_task = cursor.fetchall()
+
+    if len(deleted_task) == 0:
+        return make_response(jsonify({"description": f"Task '{task_id}' not found"}), 404)
+    else:
+        statement = "DELETE FROM TASKS WHERE task_id = ?"
+        cursor.execute(statement, (task_id,))
+        connection.commit()
+        cursor.close()
+        connection.close()
+        return make_response(jsonify({"description": f"Task '{task_id}' successfully deleted"}), 200)
+
+
+def update_single_todo(post_request, task_id):
+    mutable_tasks_file = ("title", "description", "status", "do_date")
+    change_tasks_file = []
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    status_in_request = "status" in post_request.keys()
+
+    if not task_id_exist(task_id):
+        return make_response(jsonify({"description": f"Task id '{task_id}' doesn't exist"}), 404)
+
+    if status_in_request:
+        if not status_correct(post_request["status"]):
+            return make_response(jsonify({"description": "Wrong request, status doesn't exist"}), 404)
+
+    # check if request keys are correct
+    for key in post_request:
+        if key in mutable_tasks_file:
+            change_tasks_file.append(key)
+
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+
+    # change in database
+    for file in change_tasks_file:
+        statement = f"UPDATE TASKS SET {file} = ? WHERE task_id = ?"
+        cursor.execute(statement, (post_request[file], task_id))
+        connection.commit()
+    cursor.close()
+    connection.close()
+    return make_response(jsonify({"description": f"Task '{task_id}' updated successfully"}), 200)
+
+
+def get_user_todos(user_id):
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+    statement = "SELECT * FROM TASKS WHERE user_id = ?"
+    cursor.execute(statement, (user_id,))
+    matching_tasks = cursor.fetchall()
+    tasks = [tasks_response_builder(task) for task in matching_tasks]
+    if len(matching_tasks) == 0:
```

Comment:
You are returning a "Wrong user ID" error when no tasks are found for the given user, which is not necessarily correct.

Instead, you should first verify whether the user with the provided ID exists. If the user does not exist, return an appropriate error. However, if the user exists and simply has no tasks, you should return an empty array.